src/Microsoft.Net.Http.Headers/HeaderUtilities.cs

```diff
@@ -198,6 +199,89 @@ internal static bool AreEqualCollections<T>(ICollection<T> x, ICollection<T> y,
             return current;
         }
 
+        // Try to get the value of a specific header from a list of headers
+        // e.g. { "headerValue=10, targetHeaderValue=30", "headerValue2=20" }
+        public static bool TryParseTimeSpan(StringValues headerValues, string targetValue, out TimeSpan? value)
+        {
+            for (var i = 0; i < headerValues.Count; i++)
+            {
+                var headerValue = headerValues[i];
+
+                var index = headerValue.IndexOf(targetValue, StringComparison.OrdinalIgnoreCase);
+                if (index != -1)
+                {
+                    index += targetValue.Length;
+                    long seconds;
+                    if (!TryParseInt64FromHeaderValue(index, headerValue, out seconds))
+                    {
+                        break;
+                    }
+                    value = TimeSpan.FromSeconds(seconds);
+                    return true;
+                }
+            }
+            value = null;
+            return false;
+        }
+
+        public static bool Contains(StringValues headerValues, string targetValue)
+        {
+            for (var i = 0; i < headerValues.Count; i++)
+            {
+                var index = headerValues[i].IndexOf(targetValue, StringComparison.OrdinalIgnoreCase);
+                if (index != -1)
+                {
+                    return true;
+                }
+            }
+
+            return false;
+        }
+
+        private static bool TryParseInt64FromHeaderValue(int startIndex, string headerValue, out long value)
+        {
+            var found = false;
+            while (startIndex != headerValue.Length)
```

Comment:
Could potentially turn this into unsafe code too. Not sure if that makes it any faster.